lib/test/segy.cpp

```diff
@@ -1172,6 +1172,63 @@ SCENARIO( "modifying trace header", "[c.segy]" ) {
     }
 }
 
+
+
+int check_sorting( const std::string& name ) {
+    int trace0 = 3600;
+    int trace_bsize = 10 * 4;
+    int sorting;
+
+    segy_file* fp = openfile( name, "rb" );
+
+    Err err = segy_sorting( fp,
+                            SEGY_TR_INLINE,
+                            SEGY_TR_CROSSLINE,
+                            SEGY_TR_OFFSET,
+                            &sorting,
+                            trace0,
+                            trace_bsize );
+    CHECK( success( err ) );
+    segy_close( fp );
```

Comment:
If you use the unique_ptr thing around openfile, you don't need an explicit close, and you remove the possibility for a leak here,